### DISCUSSION TITLE
AMD RX 6700 XT, sudden computer reboots

### DISCUSSION THREAD

**Discussion by sigboe** [GpuZelenograd/memtest_vulkan#45] [Card-specific memtest_vulkan error logs]
I bought a used GPU been used for mining, the seller didn't talk about issues with the GPU but I got it for $200. He also didn't say it was used for mining, but it was obvious looking at how many graphics cards he was selling. I bought it about two years ago, in a dip in the market, but other similar GPU's cost about $450 locally. Including him selling other same type GPUs for $400. He guessed I wouldn't complain trying to fix it my self. One fan had a broken ball-bearing, so I got new fans from Ali Express, I got new "high performance" EK branded heatpads from a local boutique. I replaced thermal paste, and resoldered the fan connector. All super cheap fixes

I've meticulously looked through every part of the PCBA and there are no visible damage anywhere. I can only presume there is an issue with a component, and the issue is not visible. Because I've had random reboots occurring. 

I have an issue with my GPU where it suddenly shuts off the computer. And it wasn't so bad in the beginning. I used CoreCtl to down clock the gpu a bit, and it was stable for a long time. But then it started dying in games again. More often and more often.

After the computer failed to POST at all when it died. And the comptuer worked with my wifes GTX 980 (no I can't go back to 4gb ram) I decided to reflow the core by heating it to ~200C using a heatgun, and a heatshield exposing only the core. **Then it succeeds POST again!!!** Both NVMe drives containing operating systems didn't work properly, so, I decided to try a new flavor of linux. Why does that matter? Well, that the GPU has been tested in various driver versions. I've been running Arch Linux for about a decade. MESA has been fairly "fresh" close to the newest version the last two years. I feel like different versions had more or less crashes (in retrospect, that may very well be that it uses the memory slightly differently, or it may be just a coincidence). And After having to reinstall again, I tried NixOS, which by default has "stable" packages, refreshing two times per year. So now I guess I am using MESA 24.0.7. On arch I had what ever was most recent in first week of July when the computer stopped POSTing. 

So at the time of reinstalling, I tried to download a couple of games on steam again, I had no idea that the sudden reboots would still occur. Since reflowing the GPU core fixed a problem I thought had built up to preventing POST. Diablo 4 worked for several minutes, idling in game. Then I tried Darktide and I got a sudden reboot. 

I stopped using the computer for a bit, and then I wanted to try furmark, which worked for a long time no problem.... Still thinking I thought it might be memory related. So I googled about testing VRAM. And found this project. Running memtest_vulkan for a few minutes triggered a shutdown. So I ran it again redirecting the output to a file

I am guessing its a memory chip, if not, maybe one of the VRMS

```
https://github.com/GpuZelenograd/memtest_vulkan v0.5.0 by GpuZelenograd
To finish testing use Ctrl+C

1: Bus=0x0A:00 DevId=0x73DF   12GB AMD Radeon RX 6700 XT (RADV NAVI22)
2: Bus=0x00:00 DevId=0x0000   32GB llvmpipe (LLVM 17.0.6, 256 bits)
(first device will be autoselected in 3 seconds)   Override index to test:Standard 5-minute test of 1: Bus=0x0A:00 DevId=0x73DF   12GB AMD Radeon RX 6700 XT (RADV NAVI22)
      1 iteration. Passed  0.0681 seconds  written:    7.2GB 295.0GB/sec        checked:   10.9GB 249.6GB/sec
     16 iteration. Passed  1.0515 seconds  written:  108.8GB 283.7GB/sec        checked:  163.1GB 244.1GB/sec
     87 iteration. Passed  5.0018 seconds  written:  514.8GB 282.2GB/sec        checked:  772.1GB 243.0GB/sec
    517 iteration. Passed 30.0146 seconds  written: 3117.5GB 285.6GB/sec        checked: 4676.2GB 244.8GB/sec
    952 iteration. Passed 30.0232 seconds  written: 3153.8GB 289.4GB/sec        checked: 4730.6GB 247.4GB/sec
   1384 iteration. Passed 30.0604 seconds  written: 3132.0GB 286.7GB/sec        checked: 4698.0GB 245.5GB/sec
   1815 iteration. Passed 30.0468 seconds  written: 3124.8GB 286.2GB/sec        checked: 4687.1GB 245.0GB/sec
   2251 iteration. Passed 30.0435 seconds  written: 3161.0GB 290.0GB/sec        checked: 4741.5GB 247.7GB/sec
   2675 iteration. Passed 30.0006 seconds  written: 3074.0GB 281.4GB/sec        checked: 4611.0GB 241.7GB/sec
   3098 iteration. Passed 30.0034 seconds  written: 3066.8GB 280.7GB/sec        checked: 4600.1GB 241.1GB/sec
   3520 iteration. Passed 30.0000 seconds  written: 3059.5GB 280.0GB/sec        checked: 4589.2GB 240.6GB/sec
   3947 iteration. Passed 30.0354 seconds  written: 3095.8GB 283.4GB/sec        checked: 4643.6GB 243.0GB/sec
   3947 iteration. Passed 30.0354 seconds  written: 3095.8GB 283.4GB/sec        checked: 4643.6GB 243.0GB/sec
```

**Comment by galkinvv** [comment]
In general such symthoms, especially shutdowns does NOT look like VRAM errors (but some other hardware errors, maybe the power ICs on the board). Does not look like a driver issue. Driver bugs may lead to hang/crash/black screen but not shutting down.

VRAM errors during gaming can lead to a variable behaviour, but typically at least sometimes it leads to visual artifacts. Also it typically at least sometimes produces messages about memory errors logged into memtest_vulkan.log (check it,it may keep some error information written just before shutdown). If you never had such symthoms - it doesn't look like VRAM error.

Not sure about repair possibility for such case, but you may find some more aggresive performance-limiting ways to keep the GPU still usable for some more time. Start with the most extreme downclocking like
```
cd /sys/class/drm/card0/device
echo manual >  power_dpm_force_performance_level
echo 1 > pp_dpm_socclk
echo 2 > pp_dpm_fclk
echo 1 > pp_dpm_socclk
echo 1 > pp_dpm_sclk
echo 1 > pp_dpm_mclk
```
And check if this extremely limited state is stable. If it is try to release/increase some limits to find increasing what leads to instability.
(increasing freqs step-by-step may be a bit tricky, I suppose issuing commands for tuning frequency tables, not sure about this)

**Comment by sigboe** [comment]
@galkinvv Oh wow, thank you for a fast reply, I tried to issue the commands, the device was card1, I don't have any card0, I don't have an igpu (sadly, I could have used that now until the next gen cards are released)

The numbers you suggested didn't help, but 0 on all of them, did help. The `Standard 5-minute test PASSed!`, I didn't run it for hours, but at least its a start. Side effects is that while the test was running the mouse was laggy and writing in this text field was stuttery. The tests were about 1/10 of the speed of the normal test. I suppose I will try to replicate this using CoreCtrl, and find something stable. If I can't then I will look at how to control the clocks using `/sys` 

I just need to be able to hold on until new graphics cards are released, deciding between buying the next gen or the current gen for a reduced price. This time I guess I will not go for used 


For completeness with echo 0 to all the clock domains you listed above
```
root@nixos /sys/class/drm/card1/device # , memtest_vulkan
https://github.com/GpuZelenograd/memtest_vulkan v0.5.0 by GpuZelenograd
To finish testing use Ctrl+C

1: Bus=0x0A:00 DevId=0x73DF   12GB AMD Radeon RX 6700 XT (RADV NAVI22)
2: Bus=0x00:00 DevId=0x0000   32GB llvmpipe (LLVM 17.0.6, 256 bits)
(first device will be autoselected in 0 seconds)   Override index to test:
    ...first device autoselected
Standard 5-minute test of 1: Bus=0x0A:00 DevId=0x73DF   12GB AMD Radeon RX 6700 XT (RADV NAVI22)
      1 iteration. Passed  0.8581 seconds  written:    7.2GB  17.1GB/sec        checked:   10.9GB  25.1GB/sec
      3 iteration. Passed  1.7103 seconds  written:   14.5GB  17.2GB/sec        checked:   21.8GB  25.1GB/sec
      9 iteration. Passed  5.1337 seconds  written:   43.5GB  17.2GB/sec        checked:   65.2GB  25.1GB/sec
     45 iteration. Passed 30.7683 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
     81 iteration. Passed 30.7656 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
    117 iteration. Passed 30.7735 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
    153 iteration. Passed 30.7444 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
    189 iteration. Passed 30.7833 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
    225 iteration. Passed 30.7730 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
    261 iteration. Passed 30.7980 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
    297 iteration. Passed 30.7979 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
    333 iteration. Passed 30.8001 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
Standard 5-minute test PASSed! Just press Ctrl+C unless you plan long test run.
Extended endless test started; testing more than 2 hours is usually unneeded
use Ctrl+C to stop it when you decide it's enough
    404 iteration. Passed 30.8513 seconds  written:  261.0GB  17.1GB/sec        checked:  391.5GB  25.0GB/sec
    440 iteration. Passed 30.8137 seconds  written:  261.0GB  17.2GB/sec        checked:  391.5GB  25.1GB/sec
^C
memtest_vulkan: no any errors, testing PASSed.
  press any key to continue...
```
